packages/bundle-source/cache.js

```diff
@@ -107,13 +107,38 @@ export const makeBundleCache = (wr, cwd, readPowers, opts) => {
   };
 
   /**
-   * @param {BundleMeta} meta
+   * @param {string} targetName
+   * @param {Logger} _log
+   * @returns {Promise<string>}
+   */
+  const loadMetaText = (targetName, _log = defaultLog) =>
+    wr
+      .readOnly()
+      .neighbor(toBundleMeta(targetName))
+      .maybeReadText()
+      .catch(
+        ioError =>
+          Fail`${targetName}: cannot read bundle metadata: ${q(ioError)}`,
+      );
+
+  /**
    * @param {string} targetName
    * @param {*} rootOpt
    * @param {Logger} [log]
+   * @param {BundleMeta} [meta]
    * @returns {Promise<BundleMeta>}
    */
-  const validate = async (meta, targetName, rootOpt, log = defaultLog) => {
+  const validate = async (
```

Comment:
Ah, a wiser me would have recognized that `validate` was public API. Thanks for back-filling.